### PR TITLE
bug - Cisco NAC key error

### LINKED ISSUE #15933: NAC pool on Cisco 9200 (4 switch in stack)
### The problem

Problem with polling the nac module for Cisco 9200 switches connected in a stack (4 switches).

The problem occurs when polling more than 77 ports (Cisco 9200 switches in the stack have port_id's from 1-233).

The merit is that the parameters from the switch MIB - CISCO-AUTH-FRAMEWORK-MIB they return correct values from the switch, but with debug poller nac I get this error

```
Error polling nac module for xxx.yyy.zzz.vvv. ErrorException: Undefined array key "cafSessionTimeout" in /opt/librenms/LibreNMS/OS/Shared/Cisco.php:405
Stack trace:
#0 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#2 /opt/librenms/app/Providers/ErrorReportingProvider.php(199): call_user_func()
#3 /opt/librenms/LibreNMS/OS/Shared/Cisco.php(405): App\Providers\ErrorReportingProvider->handleError()
#4 /opt/librenms/LibreNMS/Modules/Nac.php(82): LibreNMS\OS\Shared\Cisco->pollNac()
#5 /opt/librenms/LibreNMS/Poller.php(176): LibreNMS\Modules\Nac->poll()
#6 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#7 /opt/librenms/app/Console/Commands/DevicePoll.php(45): LibreNMS\Poller->poll()
#8 /opt/librems/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#9 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#11 /opt/librems/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#12 /opt/librems/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#13 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call()
#14 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#15 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\Component\Console\Command\Command->run()
#16 /opt/librenms/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run()
#17 /opt/librenms/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#18 /opt/librenms/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#19 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#20 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#21 /opt/librenms/lnms(38): App\Console\Kernel->handle()
#22 {main}
```

### Output of ./validate.php

```txt
===========================================
Component | Version
--------- | -------
LibreNMS  | 24.3.0-9-g7dd82a58b (2024-04-10T13:31:34+02:00)
DB Schema | 2024_02_07_151845_custom_map_additions (290)
PHP       | 8.2.13
Python    | 3.8.10
Database  | MariaDB 10.3.38-MariaDB-0ubuntu0.20.04.1
RRDTool   | 1.7.2
SNMP      | 5.8
===========================================

[OK]    Composer Version: 2.4.1
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQL and PHP time match
[OK]    Distributed Polling setting is enabled globally
[OK]    Connected to rrdcached
[OK]    Active pollers found
[OK]    Dispatcher Service is enabled
[OK]    Locks are functional
[OK]    Python wrapper cron entry is not present
[OK]    Redis is functional
[INFO]  fping FAILURES can be ignored if running LibreNMS in a jail without ::1. You may want to test it manually: fping ::1
[FAIL]  /usr/bin/fping6 could not be executed. /usr/bin/fping6 must have CAP_NET_RAW capability (getcap) or suid. Selinux exclusions may be required.
 (/usr/bin/fping6: can't create socket (must run as root?))
[OK]    rrdtool version ok
[OK]    Connected to rrdcached
```

### What was the last working version of LibreNMS?

24.3.0

### Anything in the logs that might be useful for us?

```txt
Error polling nac module for xxx.yyy.zzz.vvv. ErrorException: Undefined array key "cafSessionAuthUserName" in /opt/librenms/LibreNMS/OS/Shared/Cisco.php:400
Stack trace:
#0 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#2 /opt/librenms/app/Providers/ErrorReportingProvider.php(199): call_user_func()
#3 /opt/librenms/LibreNMS/OS/Shared/Cisco.php(400): App\Providers\ErrorReportingProvider->handleError()
#4 /opt/librenms/LibreNMS/Modules/Nac.php(82): LibreNMS\OS\Shared\Cisco->pollNac()
#5 /opt/librenms/LibreNMS/Poller.php(176): LibreNMS\Modules\Nac->poll()
#6 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#7 /opt/librenms/app/Console/Commands/DevicePoll.php(45): LibreNMS\Poller->poll()
#8 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#9 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#11 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#12 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#13 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call()
#14 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#15 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\Component\Console\Command\Command->run()
#16 /opt/librenms/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run()
#17 /opt/librenms/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#18 /opt/librenms/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#19 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#20 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#21 /opt/librenms/lnms(38): App\Console\Kernel->handle()
#22 {main}  
Undefined array key "cafSessionAuthUserName" {"exception":"[object] (ErrorException(code: 0): Undefined array key \"cafSessionAuthUserName\" at /opt/librenms/LibreNMS/OS/Shared/Cisco.php:400)"}
```

### PR DESCRIPTION
**Pull request by PipoCanaja**
fixes #15933

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/OS/Shared/Cisco.php [hunk 1]
```diff
@@ -396,15 +396,15 @@ public function pollNac()
...
-                    'domain' => $portAuthSessionEntryParameters['cafSessionDomain'],
-                    'username' => $portAuthSessionEntryParameters['cafSessionAuthUserName'],
-                    'ip_address' => (string) IP::fromHexString($portAuthSessionEntryParameters['cafSessionClientAddress'], true),
-                    'host_mode' => $portAuthSessionEntryParameters['cafSessionAuthHostMode'],
-                    'authz_status' => $portAuthSessionEntryParameters['cafSessionStatus'],
-                    'authz_by' => $portAuthSessionEntryParameters['cafSessionAuthorizedBy'],
-                    'timeout' => $portAuthSessionEntryParameters['cafSessionTimeout'],
-                    'time_left' => $portAuthSessionEntryParameters['cafSessionTimeLeft'],
-                    'vlan' => $portAuthSessionEntryParameters['cafSessionAuthVlan'],
+                    'domain' => $portAuthSessionEntryParameters['cafSessionDomain'] ?? '',
```
Comment (murrant) [inline LibreNMS/OS/Shared/Cisco.php:399]: These should probably be this:
```suggestion
                    'domain' => $portAuthSessionEntryParameters['cafSessionDomain'] ?? null,
```

Comment (murrant) [reply LibreNMS/OS/Shared/Cisco.php:399]: (except the IP one)

Comment (PipoCanaja) [reply LibreNMS/OS/Shared/Cisco.php:399]: Not so sure about null here. 
```
MariaDB [librenms]> describe ports_nac;
+--------------+------------------+------+-----+---------+----------------+
| Field        | Type             | Null | Key | Default | Extra          |
+--------------+------------------+------+-----+---------+----------------+
| ports_nac_id | int(10) unsigned | NO   | PRI | NULL    | auto_increment |
| auth_id      | varchar(50)      | NO   |     | NULL    |                |
| device_id    | int(10) unsigned | NO   | MUL | NULL    |                |
| port_id      | int(10) unsigned | NO   | MUL | NULL    |                |
| domain       | varchar(50)      | NO   |     | NULL    |                |
| username     | varchar(50)      | NO   |     | NULL    |                |
| mac_address  | varchar(50)      | NO   |     | NULL    |                |
| ip_address   | varchar(50)      | NO   |     | NULL    |                |
| host_mode    | varchar(50)      | NO   |     | NULL    |                |
| authz_status | varchar(50)      | NO   |     | NULL    |                |
| authz_by     | varchar(50)      | NO   |     | NULL    |                |
| authc_status | varchar(50)      | NO   |     | NULL    |                |
| method       | varchar(50)      | NO   |     | NULL    |                |
| timeout      | varchar(50)      | NO   |     | NULL    |                |
| time_left    | varchar(50)      | YES  |     | NULL    |                |
| vlan         | int(10) unsigned | YES  |     | NULL    |                |
| time_elapsed | varchar(50)      | YES  |     | NULL    |                |
| created_at   | timestamp        | YES  |     | NULL    |                |
| updated_at   | timestamp        | YES  |     | NULL    |                |
| historical   | tinyint(1)       | NO   |     | 0       |                |
+--------------+------------------+------+-----+---------+----------------+
```
Only a few of them are nullable. I'll change them

Comment (murrant) [reply LibreNMS/OS/Shared/Cisco.php:399]: bizarre with the default value there, sounds good.

**Review by murrant — APPROVED** [review]

**Pull request merged**